build/rewrite.go

```diff
@@ -888,3 +896,70 @@ func (args loadArgs) Less(i, j int) bool {
 	}
 	return args.To[i].Name < args.To[j].Name
 }
+
+// indentDocstrings modifies the value of docstrings if their indentation level has been changed
+func indentDocstrings(f *File, info *RewriteInfo) {
+	Walk(f, func(v Expr, stk []Expr) {
+		def, ok := v.(*DefStmt)
+		if !ok {
+			return
+		}
+		if len(def.Body) == 0 {
+			return
+		}
+		docstring, ok := def.Body[0].(*StringExpr)
+		if !ok || !docstring.TripleQuote {
+			return
+		}
+
+		oldIndentation := docstring.Start.LineRune - 1 // LineRune starts with 1
+		newIndentation := nestedIndentation * len(stk)
+		updatedString := indentString(docstring.Value, oldIndentation, newIndentation)
+		if updatedString != docstring.Value {
+			docstring.Value = updatedString
+			info.IndentDocstrings++
+		}
+	})
+}
+
+// indentString modifies a sting value of a docstring to match the new indentation level
+func indentString(value string, oldIndentation, newIndentation int) string {
+	if oldIndentation == newIndentation {
+		return value
+	}
+	difference := newIndentation - oldIndentation
+	lines := strings.Split(value, "\n")
+	for i, line := range lines {
+		if i == 0 {
+			// The first line shouldn't be touched because it starts right after ''' or """
+			continue
+		}
+		if difference > 0 {
+			line = strings.Repeat(" ", difference) + line
+		} else {
+			for i, rune := range line {
+				if i == -difference || rune != ' ' {
+					line = line[i:]
+					break
+				}
+			}
+		}
+		if i != len(lines)-1 && isAllSpaces(line) {
+			// Replace a line that contains only spaces with an empty line
+			// unless it's the last line that's responsible for the indentation of the closing """
+			line = ""
+		}
+		lines[i] = line
```

Comment:
As we're processing the string, we could remove trailing whitespace at the same time.

Reply:
Done. If I do that, should I also modify docstrings that are not being indented to remove trailing spaces from their lines? `"indentdocstrings"` will need to be renamed to `"formatdocstrings"` in that case.

Reply:
Remove trailing space from all docstrings for consitency.